### DISCUSSION TITLE
Mesa 3.0 examples?

### DISCUSSION THREAD

**Discussion by catherinedevlin** [mesa/mesa-examples#129] [Q&A]
It looks like all the examples are for Mesa 2.0; is there a place with examples upgraded to 3.0?  Or should there be?  A "3.0" branch, maybe?  End users don't need them yet but I kind of do for the sake of developing.

**Comment by EwoutH** [comment]
This is a good question. I think we’re going to slowly update all our example to support the latest features.

It would be useful to have tagged releases of examples that are guaranteed to work with a certain Mesa version. I briefly touched on this in https://github.com/projectmesa/mesa-examples/issues/54#issuecomment-1848924165.

@projectmesa/maintainers we probably need to revisit this.

**Comment by jackiekazil** [comment]
@EwoutH I could see this... but maybe it is a little overkill at this moment -- people can just pull previous versions which are tagged in the requirements files. 

I think I am more concerned with updating the examples.

**Comment by PRIYANSHU2026** [comment]
can i get some refference of Updated Visuvalization and how its work and perform in the agent based model

**Comment by EwoutH** [comment]
@catherinedevlin we now split of a separate branch for Mesa 2.3.x examples, and are updating examples on the main branch for Mesa 3.0. For more details, see:
- https://github.com/projectmesa/mesa-examples/issues/148

We could use help updating the examples!
